cmd/sidecars/sidecar_shim.go

```diff
@@ -90,6 +101,39 @@ func (s infoServer) Info(ctx context.Context, params *hooksInfo.InfoParams) (*ho
 
 type v1Alpha1Server struct{}
 type v1Alpha2Server struct{}
+type v1Alpha3Server struct {
+	done chan struct{}
+}
+
+func (s v1Alpha3Server) OnDefineDomain(_ context.Context, params *hooksV1alpha3.OnDefineDomainParams) (*hooksV1alpha3.OnDefineDomainResult, error) {
+	log.Log.Info(onDefineDomainLoggingMessage)
+	newDomainXML, err := runOnDefineDomain(params.GetVmi(), params.GetDomainXML())
+	if err != nil {
+		log.Log.Reason(err).Error("Failed OnDefineDomain")
+		return nil, err
+	}
+	return &hooksV1alpha3.OnDefineDomainResult{
+		DomainXML: newDomainXML,
+	}, nil
+}
```

Comment:
This was asked before, this is my answer:
https://github.com/kubevirt/kubevirt/pull/9231#discussion_r1255407468

Instead, the path forward is deprecating old v1alpha1 and removing it from the codebase and keep moving the tests to a more recent version. This is something I plan to propose later on as well.

Reply:
> Instead, the path forward is deprecating old v1alpha1 and removing it from the codebase and keep moving the tests to a more recent version. This is something I plan to propose later on as well.

Makes sense. And since it's alpha version API, we can even deprecate all but last? Anyway, agree on doing this later. 👍🏽